pkg/kubernetes/scripts/kube-client-cockpit.js

```diff
@@ -829,8 +829,17 @@
                 }
 
                 if (user) {
-                    if (user.token)
-                        options.headers["Authorization"] = "Bearer " + user.token;
+                    token = user.token;
+                    provider = user["auth-provider"] || {};
+                    if (provider.config) {
+                        if (provider.config['access-token'])
+                            token = provider.config['access-token'];
+                        else if (provider.config['token'])
+                            token = provider.config['token'];
+                    }
```

Comment:
With this the config-provided token will be preferred over `user.token` even if the latter exists. Is this intended? Could this cause some backwards compatibility issues?

Reply:
A user shouldn't have both according the spec. But in cases that they do, this would match the kubectl behavior so i think it's correct.